### PR TITLE
List snapshot

### LINKED ISSUE #382: Implement TableAdmin::ListSnapshots
Function to list all the available snapshots. #329

### PR DESCRIPTION
**Pull request by paprat**
This fixes #382.

### CODE REVIEW

File: bigtable/client/internal/table_admin.cc [hunk 1]
```diff
@@ -193,6 +193,42 @@ bool TableAdmin::CheckConsistency(
   return response.consistent();
 }
 
+std::vector<btproto::Snapshot> TableAdmin::ListSnapshots(
```
Comment (coryan) [inline bigtable/client/internal/table_admin.cc]: I wonder if this should be an iterator type.  If we expect this to return thousands to maybe a million entries then a `std::vector<>` (or another collection) is probably ideal.  If we expect millions to billions then we need an iterator.  @ghaisa and @mbrukman thoughts?

File: bigtable/client/internal/table_admin.cc [hunk 2]
```diff
@@ -193,6 +193,42 @@ bool TableAdmin::CheckConsistency(
...
+    int32_t page_size, grpc::Status& status,
```
Comment (coryan) [inline bigtable/client/internal/table_admin.cc]: We use `std::int32_t`, the fact that protobuf and grpc lazily pollute the user namespace does not mean that we should.

File: bigtable/client/internal/table_admin.h [hunk 3]
```diff
@@ -114,6 +114,10 @@ class TableAdmin {
                         bigtable::ConsistencyToken const& consistency_token,
                         grpc::Status& status);
 
+  std::vector<::google::bigtable::admin::v2::Snapshot> ListSnapshots(
```
Comment (coryan) [inline bigtable/client/internal/table_admin.h]: Consider making the returned collection optional.  I know we hardcoded vector elsewhere, we should fix those too.

**Comment by coryan** [comment]
@mbrukman and @ghaisa any thoughts on how many snapshots we expect customers to keep?  The API changes if we think it is going to be more than 100,000 or so.

**Comment by coryan** [comment]
Just talked to @mbrukman and @ghaisa, we can load the results into memory.

**Comment by coryan** [comment]
@paprat You need to rebase and address the comments I left earlier.

File: bigtable/client/internal/table_admin.h (hunk 3 above)
Comment (paprat) [reply bigtable/client/internal/table_admin.h]: As far as I know, std::optional was introduced in c++17. Do you still want me to add this?

Comment (coryan) [reply bigtable/client/internal/table_admin.h]: I meant optional as we did for SampleRowKeys.

**Pull request merged**